src/web/mjs/connectors/MangaParkEN.mjs

```diff
@@ -10,43 +9,113 @@ export default class MangaParkEN extends Connector {
         super.label = 'MangaPark';
         this.tags = [ 'manga', 'multi-lingual' ];
         this.url = 'https://mangapark.net';
+        this.apiURL = `${this.url}/apo/`;
+
+        this.requestOptions.headers.set('x-origin', this.url);
+        this.requestOptions.headers.set('x-referer', `${this.url}/`);
         this.requestOptions.headers.set('x-cookie', 'set=h=1;');
     }
 
     async _getMangaFromURI(uri) {
         const request = new Request(uri, this.requestOptions);
-        const queryMangaTitle = /\/title\/\d+/.test(uri.pathname) ? 'main h3' : 'meta[property="og:title"]';
+        const queryMangaTitle = /\/title\/\d+/.test(uri.pathname) ? 'main h3 a' : 'meta[property="og:title"]';
         const data = await this.fetchDOM(request, queryMangaTitle);
-        return new Manga(this, uri.pathname.match(/\/(\d+)\/?/)[1], (data[0].textContent || data[0].content).trim());
+        const id = parseInt(uri.pathname.match(/\/(\d+)\/?/)[1]);
+        const title = (data[0].textContent || data[0].content).trim();
+        return new Manga(this, id, title);
     }
 
     async _getMangas() {
+        try {
+            return await this._getMangasV3();
+        } catch (error) {
+            return this._getMangasV5();
```

Comment:
Miss "await" on _getMangasV5 call i think.

Reply:
Technically these methods can return a promise, because they are called with await. But yes, it's better to return an array. Thanks again. :)